agent/ebs/watcher.go

```diff
@@ -0,0 +1,212 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ebs
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/statechange"
+	apiebs "github.com/aws/amazon-ecs-agent/ecs-agent/api/resource"
+	log "github.com/cihub/seelog"
+)
+
+type EBSWatcher struct {
+	ctx        context.Context
+	cancel     context.CancelFunc
+	scanTicker *time.Ticker
+	agentState dockerstate.TaskEngineState
+	// TODO: The dataClient will be used to save to agent's data client as well as start the ACK timer. This will be added once the data client functionality have been added
+	// dataClient     data.Client
```

Comment:
Yep, this will most likely be updated/modified when we get to the attachment handler/responders where we'll actually hook up and start the watcher when agent starts.

Reply:
Added a comment for both ebsChangeEvent and dataClient as TODO. Will be used in very near future PR.